src/flowMC/strategy/adapt_step_size.py

```
@@ -0,0 +1,152 @@
+            f"Resource {self.state_name} must be a State"
+        )
+
+        # Get the acceptance buffer name from state
+        assert isinstance(
+            buffer_name := state.data.get(self.acceptance_buffer_key), str
+        ), (
+            f"State key {self.acceptance_buffer_key} must point to a string "
+            f"(buffer name)"
+        )
+
+        assert isinstance(acceptance_buffer := resources[buffer_name], Buffer), (
```

Ruff (0.14.13)

Avoid assignment expressions in `assert` statements

(RUF018)